src/graph.rs

```diff
@@ -378,6 +378,73 @@ impl PyGraph {
         Ok(data)
     }
 
+    /// Return the list of edge indices incent to a provided node
+    ///
+    /// :param int node: The node index to get incident edges from. If
+    ///     this node index is not present in the graph this method will
+    ///     return an empty list and not error.
+    ///
+    /// :returns: A list of the edge indices incident to a node in the graph
+    /// :rtype: EdgeIndices
+    pub fn incident_edges(&self, node: usize) -> EdgeIndices {
+        EdgeIndices {
+            edges: self
+                .graph
+                .edges(NodeIndex::new(node))
+                .map(|e| e.id().index())
+                .collect(),
+        }
+    }
```

Comment:
For consistency reasons I am against adding this method, as there is no equivalent on `PyDiGraph`.

I would rather have us implement https://github.com/Qiskit/retworkx/blob/main/src/digraph.rs#L1506-L1526 and https://github.com/Qiskit/retworkx/blob/main/src/digraph.rs#L1528-L1547 for consistency. That way we have the same API on both PyGraph and PyDiGraph.

Reply:
Sure I'll add duals of those `PyDiGraph` methods here. But I'll add this to `PyDiGraph` too because I prefer the index return. I'd like to get the edge index based api a bit more up to snuff.

Reply:
Done in: https://github.com/Qiskit/retworkx/pull/538/commits/3b4ae559fbe15f63de4c7d84e84d3ac1d30867e7